### ISSUE TITLE
Fallback reflectors used when script is run on startup

### ISSUE THREAD

**Issue by lrussell887** [lynxthecat/CAKE-autorate#31]
I've noticed that if the service is enabled and my router is rebooted, you can see in htop that it uses the Quad9 reflectors instead of Cloudflare. Could it be that the script is started before the network is up, causing them to be considered bad?

![putty_8VXx9dJOUb](https://user-images.githubusercontent.com/7494773/172020127-3a4f0416-3b15-486d-b5da-dbf9a95953f3.png)

**Comment by lynxthecat** [comment]
Hmm, interesting. Yes I suspect the code rotates out to the Quad9 reflector because of some form of startup-related issue.

So the code already included a check to see whether the interface existed and if not to wait until it came up. But I presume in your specific case both interfaces come up before you are able to send out and receive ping responses?

In any case, to address the issue you raised I have now added an optional startup wait in seconds. This is set in the config file here:

https://github.com/lynxthecat/CAKE-autorate/blob/fff063b3747ace8254aa090afd144fc2b0fae9c1/config.sh#L52

And now the code waits at startup before proceedings for the number of seconds set here:

https://github.com/lynxthecat/CAKE-autorate/blob/fff063b3747ace8254aa090afd144fc2b0fae9c1/CAKE-autorate.sh#L445

Please can you try experimenting with setting this value to see if you can avoid the unnecessary reflector rotation on startup? I imagine something like '120' (2 minutes) might be enough?

**Comment by lynxthecat** [comment]
By the way I'd be very interested to know how you find the performance of the script in general on your connection. Does it work well? There is a pretty active thread here: https://forum.openwrt.org/t/cake-w-adaptive-bandwidth/108848

**Comment by lrussell887** [comment]
Thank you, that startup delay I'm sure will work. The next time I can reset my router I'll give it a try. I wonder if it could be that even though the interface has come up, that the router hasn't yet gotten a DHCP lease from the modem, so there's still no connection for a few more seconds.

My connection is probably a little bit outside of the intended use case, it's a 500/50 cable connection, but I've noticed at peak times that my ISP can't provide the full speed so if I set SQM statically I end up running into occasional Bufferbloat anyway. I ended up setting the min/base to 250/25 and the max to 500/50 under the assumption that I should always be able to get half the connection speed. If you want, [here's a video](https://www.youtube.com/watch?v=hzpzhfaxIhk) of me pushing it to the extreme. 🫠 

I do have a friend I helped with getting the script setup on his router. He has a DSL connection that is only about 3.5/0.7 that is rather unpredictable due to line noise. From what I can tell it's made a notable improvement in the quality of the connection, he'll often be in a VoIP call while downloading something and you'll only hear the occasional hiccup now, and probably due to just a lack of bandwidth. I should mention that I had noticed the same startup behavior on his router, it falling back to Quad9. When I get the chance I'll see if the script can be updated there too.
